### ISSUE TITLE
@ sigil erroneous unused warning.

### ISSUE THREAD

**Issue by ShaneEverittM** [intellij-rust/intellij-rust#6626]
## Environment

* **IntelliJ Rust plugin version:** 0.3.138.3572-203
* **Rust toolchain version:** 1.49.0 (e1884a8e3 2020-12-29) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2020.3.1 (CL-203.6682.181)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** old

## Problem description
Matching on a reference to an option and using a sigil to bind to a reference to None yields an unused variable error on the None keyword in the match arm. Since None isn't variable, but a keyword, I'm pretty sure this is an error.

## Steps to reproduce
Error:

![image](https://user-images.githubusercontent.com/47866566/103715478-95ec6e80-4f75-11eb-9da6-c360635911a9.png)

Code:
```rust
fn main() {
    let mut opt = None;
    match &mut opt {
        empty @ None => *empty = Some(5),
        Some(value) => *value = 10,
    }
    dbg!(opt);
}
```
